gadopt/free_surface_equation.py

```diff
@@ -0,0 +1,58 @@
+from .equations import BaseTerm, BaseEquation
+from firedrake import dot, FacetNormal
+r"""
+This module contains the free surface terms and equations
+
+NOTE: for all terms, the residual() method returns the residual as it would be on the RHS of the equation, i.e.:
+
+  dq/dt = \sum term.residual()
+
+This sign-convention is for compatibility with Thetis' timeintegrators. In general, however we like to think about
+the terms as they are on the LHS. Therefore in the residual methods below we assemble in F as it would be on the LHS:
+
+  dq/dt + F(q) = 0
+
+and at the very end "return -F".
+"""
+
+
+class FreeSurfaceTerm(BaseTerm):
+    r"""
+    Free Surface term: u \dot n
+    """
+    def residual(self, test, trial, trial_lagged, fields, bcs):
+        assert 'free_surface_id' in self.term_kwargs
```

Comment:
```suggestion
```

The next line would raise a `KeyError` anyway.

Reply:
done!